apps/github-stack/src/github-ineraction/domain/scheduler/github-scheduler.ts

```diff
@@ -44,18 +48,25 @@ export class GitHubScheduler {
       console.log(
         `For repository ${gitRepository.full_name} latest reliase is not found`
       );
+      console.error(error);
     }
   }
 
   async checkForUpdates() {
     const repositories = await this.gitRepository.find({ relations: ["user"] });
     for (const repo of repositories) {
-      const release = await this.getLatestReliase(repo);
-      if (repo.latestRelease != release) {
-        repo.latestRelease = release;
-        this.gitRepository.save(repo);
-        await this.sendingNotification(repo);
-      }
+      const latestRelease = await this.getLatestReliase(repo);
+      repo.releases?.forEach((release) => {
```

Comment:
Have you tested this when you already have few releases for a repo? 
It seems that this logic will send notificaiton for each release that is not latest release.
I think we first need to get latest realiase of repo from database and compare it with latest release from github.

First write a test for this. And then refactor like i mentioned above, when the test passes you will know you *probably* implemented correct logic. (if test itself is correct that is)